pkg/deploy/vmsscleaner/clean.go

```diff
@@ -0,0 +1,50 @@
+package vmsscleaner
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
+)
+
+type Interface interface {
+	RemoveFailedNewScaleset(ctx context.Context, rgName, vmssName string) (deleted bool)
+}
+
+type cleaner struct {
+	log  *logrus.Entry
+	vmss compute.VirtualMachineScaleSetsClient
+}
+
+func New(log *logrus.Entry, vmss compute.VirtualMachineScaleSetsClient) Interface {
+	return &cleaner{
+		log:  log,
+		vmss: vmss,
+	}
+}
+
+func (c *cleaner) RemoveFailedNewScaleset(ctx context.Context, rgName, vmssName string) (deleted bool) {
+	scalesets, err := c.vmss.List(ctx, rgName)
+	if err != nil || len(scalesets) < 2 { // Don't remove scaleset if it's the only one
```

Comment:
Would you please explain the logic behind this? If there is only one scaleset end, but if there are 2 and more, delete all of them?
I would humbly ask for explanation comment added to this line.

Reply:
It only deletes the scaleset with the name passed in, `vmssName`. Since there can be only one VMSS with a given name, it can therefore only delete a single scaleset. It iterates over the list it already has to avoid an unnecessary call to `Get`.
I will add a comment to the function explaning its purpose — after all, it is exported.

Reply:
However, I have realized that this approach doesn't account for the case in which the VMSS has not yet been created, in which case it should not prevent a retry... I think we will need to return true by default.